src/main/java/org/folio/processing/mapping/defaultmapper/processor/Processor.java

```diff
@@ -119,12 +118,28 @@ private void handleRecordDataFieldByField(RuleExecutionContext ruleExecutionCont
     if (mappingEntry == null) {
       return;
     }
+    Map<Integer, JsonObject> fieldMappingIndicatorsSet = getFieldMappingIndicators(mappingEntry);
 
     //there is a mapping associated with this marc field
     for (int i = 0; i < mappingEntry.size(); i++) {
       //there could be multiple mapping entries, specifically different mappings
       //per subfield in the marc field
       JsonObject subFieldMapping = mappingEntry.getJsonObject(i);
+      //check if mapping entry has indicators sets
+      if (Collections.frequency(fieldMappingIndicatorsSet.values(), null) != fieldMappingIndicatorsSet.size()) {
+        JsonObject subFieldMappingEntity = (JsonObject) ((JsonArray) (subFieldMapping.getValue("entity"))).getValue(0);
```

Comment:
@bisecomsergiy  Please extract "entity", "indicators", "*" to the constants.

Reply:
Thanks Volodymyr, it is done!